plugins/module_utils/network/sonic/config/interfaces/interfaces.py

```diff
@@ -588,197 +493,127 @@ def get_interface_requests(self, configs, have):
-            if conf.get('speed') == 'SPEED_DEFAULT':
-                method = DELETE
-                eth_url = (url + '/openconfig-if-ethernet:ethernet/config/port-speed') % quote(intf_name, safe='')
-                request = {"path": eth_url, "method": method}
+        for conf in want:
+            name = conf['name']
+            intf = next((e_intf for e_intf in have if name == e_intf['name']), None)
+            if name.startswith('Loopback'):
+                if not intf:
+                    commands_add.append({'name': name})
+
+            temp_conf = dict()
+
+            temp_conf['name'] = conf['name']
+            non_eth_attribute = ('description', 'mtu', 'enabled')
+            eth_attribute = ('description', 'mtu', 'enabled', 'auto_negotiate', 'speed', 'advertised_speed', 'fec')
+            attribute = eth_attribute
+
+            if not name.startswith('Eth'):
+                attribute = non_eth_attribute
+            for attr in attribute:
+                default_val = attributes_default_value[attr]
+                if attr == 'speed':
+                    default_val = retrieve_default_intf_speed(self._module, conf['name'])
+                temp_conf[attr] = default_val if conf.get(attr, None) is None else conf[attr]
+
+            if not intf:
+                commands_add.append(temp_conf)
+                if cur_state == "overridden":
+                    delete_all = True
             else:
-                method = PATCH
-                eth_conf['port-speed'] = 'openconfig-if-ethernet:' + conf['speed']
-                eth_url = (url + '/openconfig-if-ethernet:ethernet/config') % quote(intf_name, safe='')
-                payload = {'openconfig-if-ethernet:config': eth_conf}
-                request = {"path": eth_url, "method": method, "data": payload}
-
-        return request
-
-    def build_create_autoneg_request(self, conf, have_conf):
-        intf_name = conf['name']
-        eth_conf = dict()
-        request = dict()
-        method = PATCH
-
-        if intf_name.startswith('Ethernet'):
-            eth_conf = dict()
-            if conf.get('auto_negotiate') is not None:
-                if conf.get('auto_negotiate'):
-                    eth_conf['auto-negotiate'] = True
-                else:
-                    eth_conf['auto-negotiate'] = False
-
-            c_ads = conf.get('advertised_speed', [])
-            if c_ads:
-                h_ads = have_conf.get('advertised_speed', [])
-                if h_ads is None:
-                    h_ads = []
-                new_ads = h_ads + c_ads
-                eth_conf['openconfig-if-ethernet-ext2:advertised-speed'] = ','.join(new_ads)
+                is_change = False
+                for attr in temp_conf:
+                    if attr == "advertised_speed" and temp_conf[attr]:
+                        temp_conf[attr].sort()
+
+                    if intf[attr] and intf[attr] != temp_conf[attr]:
+                        is_change = True
```

Comment:
Actually I already discussed the similar query with Kerry for l3_interface module and he agreed upon the case for replace.
https://github.com/ansible-collections/dellemc.enterprise_sonic/pull/241#issuecomment-1788504847

Reply:
That exception was only for the IPv4 and IPv6 address attributes for L3 Interfaces. It does not apply to all attributes for all resource modules. In general (and probably also for future versions of our L3 Interfaces module as well), we should be using the "replaced" handling specification that was previously distributed to and agreed to by the people in our group. Mingjun's comments here are consistent with our specification.

Please make the changes here needed to bring this section of the code into compliance with our specification.

Reply:
Adding more detail to my comment:

The argspec for the "interfaces" resource module is almost "flat", but not quite. Because the "advertised_speed" attribute is a list, we need to support "replaced" state handling to replace the content of this list on an interface without wiping out the remaining interface configuration. 

For handling of the other attributes, which are all "top level" attributes in the argspec, the current handling is acceptable. But additional logic is required to handle the special case of replacement of only the "advertised_speed" list.

Reply:
The revised handling looks okay to me.

I would like MIngjun to check the updated diff to verify that it also addresses his concerns before we mark this issue "Resolved".

Reply:
Looks good!